### PR TITLE
Ignoring file that are generated during talawa testing and firebase initialization closes #1573 and Documentation for lib/main.dart  and Fixing the TypeError in Linting #1654

### LINKED ISSUE #1573: Modify authorised list and .gitignore accordingly
Yes. The flutter file should be in `.gitignore` for now. It's more likely that many more people will reproduce this mistake than those who purposefully modify the file.

_Originally posted by @palisadoes in https://github.com/PalisadoesFoundation/talawa/issues/1563#issuecomment-1445281717_

### PR DESCRIPTION
**Pull request by SiddheshKukade**
**What kind of change does this PR introduce?**

1. Bugfix
2. Documentation for `main.dart`
3. Fixing typeerror during linting.

**Issue Number:**

Fixes #1573 #1654

**Did you add tests for your changes?**

Not required.

**Snapshots/Videos:**

image

**Summary**
These two files we're getting generated after running the `flutter test` command 
1. The `genhtml.perl` file contains current code coverage and due to that it is not relevant to merge it into the repository.
image
4. The ` test_img.png ` is a image generate to test some features of the app. Hence it is also not relevant to merge.
image

**Does this PR introduce a breaking change?**
No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by SiddheshKukade** [comment]
The code is ready to be reviewed.

**Comment by palisadoes** [comment]
What about `firebase_options.dart` does that need to be included?

**Comment by palisadoes** [comment]
We have a new check in place. All Talawa mobile app methods, classes and functions need to have suitable comments. Please refer to this issue as  reference for what is now required. The aim is to help us improve the quality of our automated documentation on docs.talawa.io.
 
- https://github.com/PalisadoesFoundation/talawa/issues/1479

To fix this issue:

1. You will be required to remove a line containing `ignore_for_file`
1. After doing so you'll be requested to update the comments appropriately

We have also tried to make the documentation step automatically update your IDE with a custom Linter. If you have any comments, please contact @literalEval who worked on this code.

**Comment by SiddheshKukade** [comment]
> What about `firebase_options.dart` does that need to be included?

Sir for giving notifications to user which were added in #1327  we require that `firebase_options.dart`  file
But what I had observed previously was when new contributors were setting up their project they did not setup the firebase project and due to that the app causes a build err with the message that `firebaseOptions is not defined`
to solve that they  just comment out those functions (to get the app running) and add that commented code in PR without even noticing it. So I think adding a try catch at `main.dart` could prevent this from happening.

```dart
  try {
    if (urlBox.get('url') != null) {
      await setUpFirebaseKeys();

      await setUpFirebase();
      await setUpFirebaseMessaging();
    }
  } catch (e) {
    print("Firebase not working");
  }
```
Commit  : https://github.com/PalisadoesFoundation/talawa/pull/1632/commits/1a4934ece63a60a9e1a21da7c04b91c2b7ff8817

**Comment by SiddheshKukade** [comment]
This check fails due to change in `main.dart` file and can be ignored here.
image

**Comment by literalEval** [comment]
@SiddheshKukade which branch do you use for pushing new commits ? I observed that all of your PRs tend to have the same list of commits, that too in large number. Please fix that.

**Comment by palisadoes** [comment]
@SiddheshKukade  rebase your repo to the latest merge. The commit list is too long and distracting.

**Review by noman2002 — CHANGES_REQUESTED** [review]
add documentation in main.dart

File: lib/main.dart [hunk 1]
```diff
@@ -96,18 +116,29 @@ Future<void> main() async {
...
+      await setUpFirebaseKeys();
 
-    await setUpFirebase();
-    await setUpFirebaseMessaging();
+      await setUpFirebase();
+      await setUpFirebaseMessaging();
+    }
+  } catch (e) {
+    print("Firebase not working");
   }
```
Comment (SiddheshKukade) [inline lib/main.dart:128]: try catch added to avoid the firebase errors during the initial build.

File: lib/main.dart [hunk 2]
```diff
@@ -175,8 +208,9 @@ class _MyAppState extends State<MyApp> {
     }
   }
 
-  // The build method is called any time you call setState ,your widget's
-  // dependencies update, or any of the parent widgets are rebuilt.
+  /// The build method is called any time you call setState ,your widget's
+  ///
```
Comment (SiddheshKukade) [inline lib/main.dart:212]: Needed to add the extra `\n /// ` to follow the docs lint rules

**Comment by literalEval** [comment]
@SiddheshKukade can you revert your changes in `talawa_good_doc`? I have done these changes in my branch for `further enhancing talawa_lint`, along with other changes. I appreciate your effort in fixing this nasty bug, but it will be great if `talawa_lint` related changes are done in a separate and dedicated place, so that analysis of them, later on, can be easier. Thanks :)

**Comment by literalEval** [comment]
Also, you can run `flutter pub run custom_lint` locally to check for analysis on your machine. You don't need to commit to check for it. If you need help in setting up `talawa_lint`, then I'm here to help :)

**Comment by SiddheshKukade** [comment]
> 



> Also, you can run `flutter pub run custom_lint` locally to check for analysis on your machine. You don't need to commit to check for it. If you need help in setting up `talawa_lint`, then I'm here to help :)

@literalEval I know that.

The custom_lint was throwing errors for null type checking 
image
So I've tried to fix them with some type checking 


But now it is still giving and unclear error of  `/// None` as you can see here. I've already added `/// None ` but it still gives error about it.
image

**Comment by literalEval** [comment]
> > 
> 
> > Also, you can run `flutter pub run custom_lint` locally to check for analysis on your machine. You don't need to commit to check for it. If you need help in setting up `talawa_lint`, then I'm here to help :)
> 
> @literalEval I know that.
> 
> The custom_lint was throwing errors for null type checking image So I've tried to fix them with some type checking
> 
> But now it is still giving and unclear error of `/// None` as you can see here. I've already added `/// None ` but it still gives error about it. image

Add explicit `void` type to that function. It will work :)

**Comment by literalEval** [comment]
Plese confirm if it does :)

**Comment by palisadoes** [comment]
> Plese confirm if it does :)

Does the script output say that the explicit void is required? If not, it needs to be updated.

**Comment by SiddheshKukade** [comment]
It worked for that,
And what should be done here ? 
image

**Comment by SiddheshKukade** [comment]
> Does the script output say that the explicit void is required? If not, it needs to be updated.
It does not but adding that up would have save me a lot of time debugging. lol

**Comment by literalEval** [comment]
> > Plese confirm if it does :)
> 
> Does the script output say that the explicit void is required? If not, it needs to be updated.

In my fix list :)

**Comment by literalEval** [comment]
> > Does the script output say that the explicit void is required? If not, it needs to be updated.
> 
> It does not but adding that up would have been helped save to me a lot of hours debugging. lol

You could have just asked me :)

**Comment by SiddheshKukade** [comment]
@literalEval  It is also not working for the getter functions.
image
image

**Comment by literalEval** [comment]
> @literalEval It is also not working for the getter functions. image image

It is working :)
A getter returns something. It's not void.
Set the return type to `String` instead of None :)

**Comment by literalEval** [comment]
I am also thinking to exclude getters and setters from `function/method` type, as the argument and return types are obvious.

**Comment by literalEval** [comment]
@SiddheshKukade that would be `Future<void>` and not void :)
And so will be the 
```
/// returns:
/// * `Future<void>`: ...

**Comment by literalEval** [comment]
> @SiddheshKukade can you revert your changes in `talawa_good_doc`? I have done these changes in my branch for `further enhancing talawa_lint`, along with other changes. I appreciate your effort in fixing this nasty bug, but it will be great if `talawa_lint` related changes are done in a separate and dedicated place, so that analysis of them, later on, can be easier. Thanks :)

@SiddheshKukade :)

**Comment by SiddheshKukade** [comment]
@noman2002, the code is ready to be reviewed. 
This error can be ignored as it is caused due to change in `.gitignore` file.
image

**Comment by palisadoes** [comment]
@literalEval Create an issue to update the script

**Comment by SiddheshKukade** [comment]
> > @SiddheshKukade can you revert your changes in `talawa_good_doc`? I have done these changes in my branch for `further enhancing talawa_lint`, along with other changes. I appreciate your effort in fixing this nasty bug, but it will be great if `talawa_lint` related changes are done in a separate and dedicated place, so that analysis of them, later on, can be easier. Thanks :)
> 
> @SiddheshKukade :)

Hello @literalEval, I've created an issue to keep report of this now.

**Comment by literalEval** [comment]
@SiddheshKukade works. Thanks.

**Comment by SiddheshKukade** [comment]
@noman2002, @TheHazeEffect, @tonythegr8 please review.

**Review by palisadoes — APPROVED** [review]

**Pull request merged**